src/main/java/uk/gov/pay/ledger/agreement/dao/AgreementDao.java

```diff
@@ -130,6 +148,18 @@ public Long getTotalForSearch(AgreementSearchParams searchParams) {
         });
     }
 
+    // Includes events for all associated payment instruments, including old payment instruments that have been replaced.
+    public List<Event> findAssociatedEvents(AgreementEntity agreementEntity, String serviceId) {
```

Comment:
Can we just pass agreement external ID as entity entity object is not needed

Reply:
Yes, that makes sense.  I am still going to look up the entity so that we can return a 404 if it's not found, but we don't need to pass the whole thing to the DAO.